src/Aspire.Hosting.Dapr/IDistributedApplicationComponentBuilderExtensions.cs

```diff
@@ -30,6 +30,7 @@ public static class IDistributedApplicationResourceBuilderExtensions
     /// <param name="builder">The resource builder instance.</param>
     /// <param name="options">Options for configuring the Dapr sidecar, if any.</param>
     /// <returns>The resource builder instance.</returns>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0027:API with optional parameter(s) should have the most parameters amongst its public overloads", Justification = "<Pending>")]
     public static IResourceBuilder<T> WithDaprSidecar<T>(this IResourceBuilder<T> builder, DaprSidecarOptions? options = null) where T : IResource
```

Comment:
Better to just have another overload that does not take the options?

Reply:
Yes, that is the norm in the ASP.NET Core codebase - we did the same for Service Discovery (2 overloads). Make the options non-nullable, too.